katsdpimager/profiling.py

```diff
@@ -229,77 +233,107 @@ def profile_device(self, queue: katsdpsigproc.abc.AbstractCommandQueue,
         start_event = queue.enqueue_marker()
         yield
         stop_event = queue.enqueue_marker()
-        self.device_records.append(DeviceRecord(frame, start_event, stop_event))
-
-    def write_csv(self, filename: Union[str, pathlib.Path]) -> None:
```

Comment:
Is this obsolete now?

Reply:
It was never called from anywhere, and I don't have any tools to consume it. In theory it could move into the CollectProfiler, but I'd rather wait until I have a use case and then probably write a new Profiler subclass that answers the specific question I have.